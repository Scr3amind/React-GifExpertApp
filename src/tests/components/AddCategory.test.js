import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {target : {value}});

        expect(wrapper.find('p').text().trim()).toBe(value);
    })

    test('no debe de enviar la informaciÃ³n con onSubmit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    })

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola mundo';

        // Simular inputChange
        input.simulate('change', {target : {value}});
        // Simular submit
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        
        //setCategories debe llamarse
        expect(setCategories).toHaveBeenCalled();
        //caja de texto limpia
        expect(input.prop('value')).toBe('');


    })
    
    
    
    
})
