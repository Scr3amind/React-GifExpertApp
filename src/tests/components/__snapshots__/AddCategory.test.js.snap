// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    className="form__field"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
